lib/plugins/chest.js

```diff
@@ -12,7 +12,9 @@ function inject (bot, { version }) {
       assert.ok(chestToOpen.type === mcData.blocksByName.chest.id ||
                 chestToOpen.type === mcData.blocksByName.ender_chest.id ||
                 chestToOpen.type === mcData.blocksByName.trapped_chest.id ||
-                (mcData.blocksByName.barrel && chestToOpen.type === mcData.blocksByName.barrel.id))
+                (mcData.blocksByName.barrel && chestToOpen.type === mcData.blocksByName.barrel.id) ||
+                mcData.blocks[chestToOpen.type].name.endsWith('_shulker_box'))
```

Comment:
If I'm not wrong, it should do something similar as with the barrel (check that the barrel does exist in mcData.blocksByName), otherwise in versions where the shulker box doesn't exist, it would throw saying that undefined.name doesn't exist.

Reply:
I did actually have this there in an earlier version, but someone beat me to a PR, which omitted it. I followed their lead and got rid of it in my version, since I figured they knew better. Shulker boxes were added in version 1.1 (about 3 years ago). Does mineflayer work for versions older than that? If so, then it should be fixed.

Reply:
Yes, it suports versions since 1.8

Reply:
I think the above implementation indirectly addresses that, as the block name cannot end with "_shulker_box" unless shulker boxes exist. So this text should work correctly in all older versions.

Reply:
I think you're right, sorry!
I got confused thinking that we were always looking for the id of the shulker box

Reply:
then I mark this comment as resolved